### PR TITLE
Added ReduceSum to ONNX importer

### PR DESCRIPTION
**Pull request by l-bat**
resolves: #16235
merge with extra: opencv/opencv_extra#792

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

<cut/>

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.2.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -387,26 +387,36 @@ void ONNXImporter::populateNet(Net dstNet)
...
             layerParams.set("ave_pool_padded_area", framework_name == "pytorch");
         }
-        else if (layer_type == "GlobalAveragePool" || layer_type == "GlobalMaxPool" || layer_type == "ReduceMean")
+        else if (layer_type == "GlobalAveragePool" || layer_type == "GlobalMaxPool" ||
+                layer_type == "ReduceMean" || layer_type == "ReduceSum")
         {
             CV_Assert(node_proto.input_size() == 1);
             layerParams.type = "Pooling";
-            layerParams.set("pool", layer_type == "GlobalMaxPool"? "MAX" : "AVE");
+            String pool =  layer_type == "GlobalMaxPool" ? "MAX" : layer_type == "ReduceSum" ? "SUM" : "AVE";
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:395]: Please change to `if else` blocks

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 2]
```diff
@@ -426,13 +436,12 @@ void ONNXImporter::populateNet(Net dstNet)
                     avgLp.name = layerParams.name + "/avg";
                     avgLp.type = "Pooling";
                     CV_Assert(layer_id.find(avgLp.name) == layer_id.end());
-                    avgLp.set("pool", "ave");
+                    avgLp.set("pool", layer_type == "ReduceMean" ? "ave": "sum");
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:439]: please reuse `String pool`

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 3]
```diff
@@ -426,13 +436,12 @@ void ONNXImporter::populateNet(Net dstNet)
...
                     if (axes.size() == 2)
                     {
-                        CV_CheckEQ(axes.get<int>(0), 1, "Unsupported ReduceMean mode");
-                        CV_CheckEQ(axes.get<int>(1), 2, "Unsupported ReduceMean mode");
+                        CV_CheckEQ(clamp(axes.get<int>(0), inpShape.size()), 1, "Unsupported ReduceMean or ReduceSum mode");
+                        CV_CheckEQ(clamp(axes.get<int>(1), inpShape.size()), 2, "Unsupported ReduceMean or ReduceSum mode");
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:443]:
```cpp
"Unsupported " + layer_type  + " mode")
```

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 4]
```diff
@@ -983,11 +997,20 @@ void ONNXImporter::populateNet(Net dstNet)
...
                     CV_Error(Error::StsNotImplemented, "Constant multiply with different shapes");
 
                 Mat out;
-                if (isDiv)
+                if (inp1.total() == 1)
+                {
+                    if (inp1.type() == CV_32F)
+                        out = (isDiv ? 1.0 / inp1.at<float>(0) : inp1.at<float>(0)) * inp0;
+                    else if (inp1.type() == CV_32S)
+                        out = (isDiv ? 1.0 / inp1.at<int>(0) : inp1.at<int>(0)) * inp0;
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:1009]: What's the problem with single element matrices?

Comment (l-bat) [reply modules/dnn/src/onnx/onnx_importer.cpp:1009]: Problem with divide or multiply for single element matrix and non-single element matrix

Comment (dkurt) [reply modules/dnn/src/onnx/onnx_importer.cpp:1015]: `inp0` or `inp1`? Can you show an example with the shapes?

Comment (l-bat) [reply modules/dnn/src/onnx/onnx_importer.cpp:1015]:
```
inp0.size = [N, 1]
inp1.size = [1]
```

Comment (dkurt) [reply modules/dnn/src/onnx/onnx_importer.cpp:1015]: Will
```cpp
out =  inp0 / inp1
```
here?

**Pull request merged**